libs/components/datetime/src/lib/modules/timepicker/timepicker.directive.ts

```diff
@@ -242,30 +257,31 @@ export class SkyTimepickerInputDirective
       }
     }
 
-    this.renderer.setProperty(
-      this.elRef.nativeElement,
+    this.#renderer.setProperty(
+      this.#elRef.nativeElement,
       'value',
       formattedValue
     );
   }
 
-  private formatter(time: any) {
+  #formatter(time: any) {
     if (time && typeof time !== 'string' && 'local' in time) {
       return time;
     }
     if (typeof time === 'string') {
       if (time.length === 0) {
         return '';
       }
-      let currentFormat: string;
+      let currentFormat: string | undefined;
       if (this.timeFormat === 'hh') {
         currentFormat = 'h:mm A';
       }
       if (this.timeFormat === 'HH') {
         currentFormat = 'H:mm';
       }
       if (typeof this.returnFormat === 'undefined') {
-        this.returnFormat = currentFormat;
+        // TODO: Remove this non-null when the typing of the `timeFormat` input is made stricter.
+        this.returnFormat = currentFormat!;
```

Comment:
Can this be refactored to only respect the two time formats? Then you wouldn't need the non-null assertion.
```
const currentFormat = (this.timeFormat === 'HH') ? 'H:mm' : 'h:mm A';

if (typeof this.returnFormat === 'undefined') {
  this.returnFormat = currentFormat;
}
```